cmd/bpf2go/makedep.go

```diff
@@ -106,3 +106,48 @@ func parseDependencies(baseDir string, in io.Reader) ([]dependency, error) {
 	}
 	return deps, nil
 }
+
+// mergeDependencies combines multiple dependency slices into one, merging prerequisites
+// for files that appear in multiple slices.
+func mergeDependencies(depsSlices ...[]dependency) []dependency {
+	// Map to track merged dependencies by file
+	merged := make(map[string][]string)
+
+	// Process each slice of dependencies
+	for _, deps := range depsSlices {
+		for _, dep := range deps {
+			// If we've seen this file before, merge prerequisites
+			if existing, ok := merged[dep.file]; ok {
+				// Combine prerequisites, avoiding duplicates
+				prereqs := make(map[string]struct{})
```

Comment:
I think you might be able to simplify this by appending to a slice. At the end, sort the slice and use https://pkg.go.dev/slices#Compact in a second pass.